apps/src/codebridge/Workspace/HeaderButtons.tsx

```diff
@@ -18,8 +31,23 @@ const WorkspaceHeaderButtons: React.FunctionComponent = () => {
   const {startSource} = useCodebridgeContext();
 
   const appName = useAppSelector(state => state.lab.levelProperties?.appName);
+  const enableMicroBit = useAppSelector(
+    state =>
+      (state.lab.levelProperties as PythonlabLevelProperties | undefined)
+        ?.enableMicroBit || false
+  );
   const skipUrl = useAppSelector(state => state.lab.levelProperties?.skipUrl);
   const dialogControl = useDialogControl();
+  const source = useAppSelector(
+    state => state.lab2Project.projectSource?.source
+  ) as MultiFileSource | undefined;
+  const files = source?.files || {};
+  let pythonCode = '';
+  for (const file of Object.values(files as object)) {
```

Comment:
We should move this logic to only occur when we are about to flash the micro:bit, otherwise we will be looping over all the files every time there is a change. We also can skip it if the lab isn't python lab (although if we move it into a function that only occurs when the button is clicked that shouldn't be an issue anymore)

Reply:
Will do!